compile/typedef.go

```diff
@@ -53,13 +55,17 @@ func compileTypedef(file string, src *ast.Typedef) (*TypedefSpec, error) {
 			Reason: err,
 		}
 	}
+	isLong := annotations["json.type"] == "Long"
+	isTimestamp := annotations["json.type"] == "Date"
```

Comment:
This isn't quite right. The annotation is not supported on the `typedef`
declaration but on the `i64`.

It's,

    typedef i64 (json.type = "Long") Timestamp

Not,

    typedef i64 Timestamp (json.type = "Long")

https://github.com/thriftrw/thriftrw-node#i64

So you should be looking at the annotations on the [I64Spec].

  [I64Spec]: https://godoc.org/go.uber.org/thriftrw/compile#I64Spec

Also, in the compile step for the I64Spec, you'll want to validate that only
valid values are provided for `json.type`. That's probably where you want to
associate this information with the I64Spec as a new `JSONType` field of type
I64JSONType`.

Reply:
I double checked our internal usage; it looks like it's all `typedef i64 (json.type = ...)`

Reply:
@abhinav ^^

Reply:
That means that abhinav is correct.

We should not read annotations on the typedef directly; we should only read it based on the `Target` annotations in the `Link()` method.

Reply:
Clarification to @Raynos suggestion: I would rather have this information tracked at I64Spec compilation time. See my original comment for the general direction this should take.